boards/riscv/icev_wireless/doc/index.rst

```diff
@@ -0,0 +1,192 @@
+brought out to J3, as well as 3 PMOD connectors for interfacing directly to
+the iCE40 FPGA. Note that several of the iCE40 pins brought out to the PMOD
+connectors are capable of operating as differential pairs.
+
+.. figure:: img/icev_wireless_back.jpg
+   :width: 400px
+   :align: center
+   :alt: ICE-V Wireless (Back)
+
+   ICE-V Wireless (Back)
+
+The J3 pins are 4V, 3.3V, NRST, GPIO2, GPIO3, GPIO8, GPIO9, GPIO10, GPIO20,
+GPIO21, FPGA_P34, and GND. Note that GPIO2 and GPIO3 may be configured for
+ADC operation.
+
+For PMOD details, please refer to the `PMOD Specification`_ and the image
+below.
+
+.. figure:: img/icev_wireless_pinout.jpg
+   :width: 400px
+   :align: center
+   :alt: ICE-V Wireless Pinout
+
+Programming and Debugging
+*************************
+
+Programming and debugging for the ICE-V Wireless ESP32-C3 target is
+incredibly easy 🎉 following the steps below.
+
+Building and Flashing
+=====================
+
+For the :code:`Hello, world!` application, follow the instructions below.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: icev_wireless
+   :goals: build flash
+
+Since the Zephyr console is by default on the `usb_serial` device, we use
+the espressif monitor to view.
+
+.. code-block:: console
+
+   $ west espressif monitor
+
+Debugging
+=========
+
+As with much custom hardware, the ESP32C3 modules require patches to
+OpenOCD that are not upstreamed. Espressif maintains their own fork of
+the project. The custom OpenOCD can be obtained by running the following extension:
+
+.. code-block:: console
+
+   west espressif install
+
+.. note::
+
+   By default, the OpenOCD will be downloaded and installed under $HOME/.espressif/tools/zephyr directory
+   (%USERPROFILE%/.espressif/tools/zephyr on Windows).
+
+The Zephyr SDK uses a bundled version of OpenOCD by default. You can overwrite that behavior by adding the
+``-DOPENOCD=<path/to/bin/openocd> -DOPENOCD_DEFAULT_PATH=<path/to/openocd/share/openocd/scripts>``
+parameter when building.
+
+Here is an example for building the :ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: icev_wireless
+   :goals: build flash
+   :gen-args: -DOPENOCD=<path/to/bin/openocd> -DOPENOCD_DEFAULT_PATH=<path/to/openocd/share/openocd/scripts>
+
+You can debug an application in the usual way. Here is an example for the
+:ref:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: icev_wireless
+   :maybe-skip-config:
+   :goals: debug
+
+Bootloader
+==========
+
+The ESP32-C3 ROM bootloader is enabled by default. This provides for a
+quick and reliable means of flashing and observing startup diagnostic
+information.
+
+Support for MCUBoot is planned 🚧 (Coming soon..) 🚧
+
+..
+   TODO: document flashing MCUBoot
+
+..
+   TODO: document flash partitions
+
+FPGA Support
+============
+
+Support for FPGA is planned 🚧 (Coming soon..) 🚧
```

Comment:
I don't really like these "Coming soon" section in the documentation. I'd suggest that we either omit this section for now and add it later, or we add support for MCUBoot/FPGA in this PR.

Reply:
@fkokosinski - thanks for sharing your preference. I would prefer not to make additional changes since there is already one approval for this PR.

We are hoping to add basic FPGA bitstream support (and additional flash partition info) in the next PR, ideally this week.